src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Operation/NewAzureVMCommand.cs

```diff
@@ -100,11 +100,22 @@ private Uri GetOrCreateStorageAccountForBootDiagnostics()
 
             if (!string.IsNullOrEmpty(storageAccountName))
             {
-                var storageAccountResponse = storageClient.StorageAccounts.GetProperties(this.ResourceGroupName, storageAccountName);
-
-                if (!storageAccountResponse.StorageAccount.AccountType.Equals(AccountType.PremiumLRS))
+                try
                 {
-                    return storageAccountResponse.StorageAccount.PrimaryEndpoints.Blob;
+                    var storageAccountResponse = storageClient.StorageAccounts.GetProperties(this.ResourceGroupName,
+                        storageAccountName);
+                    if (!storageAccountResponse.StorageAccount.AccountType.Equals(AccountType.PremiumLRS))
+                    {
+                        return storageAccountResponse.StorageAccount.PrimaryEndpoints.Blob;
+                    }
+                }
+                catch (Exception e)
+                {
+                    if (!e.Message.Contains("ResourceNotFound"))
```

Comment:
The fix is to swallow every exception during getting storage account.  If the exception is "ResourceNotFound", then the given storage account name is not in the same resource group, so no warning message is shown.  If the exception is other than "ResourceNotFound", we show a warning message, but keeps going, because we can still use other storage account for boot diagnostics.